app/models/katello/glue/pulp/repo.rb

```diff
@@ -757,8 +757,38 @@ def capsule_download_policy
         end
       end
 
+      def distributors_change?(capsule_distributors)
```

Comment:
ahhhh probably due to distributors being a lazy accessor.  What about:

distributors_matches?()

importer_matches?()